hid/joystick_base.h

```diff
@@ -179,58 +216,103 @@ class JoystickBase : public Threadable
     }
 
 protected:
-    std::array<unsigned char, toIndex(JButton::LENGTH)> m_ButtonState;
-
     JoystickBase(float deadZone = 0.0f)
       : m_DeadZone(deadZone)
     {}
 
-    void raiseEvent(JButton button, bool pressed)
+    //------------------------------------------------------------------------
+    // Declared virtuals
+    //------------------------------------------------------------------------
+    virtual bool updateState() = 0;
+
+    //------------------------------------------------------------------------
+    // Protected methods
+    //------------------------------------------------------------------------
+    void setPressed(JButton button, bool isInitial)
+    {
+        // Get state and set StateDown and StatePressed
+        uint8_t state = getState(button);
+        state |= (StateDown | StatePressed);
+
+        // Set new state
+        setState(button, state, isInitial);
+    }
```

Comment:
I don't really understand what's going on in this function. Couldn't you just write:
```
setState(button, StateDown | StatePressed, isInitial);
```
I'm a bit worried that by or-ing state you could also have ``StateReleased`` set. Or am I missing something?

Reply:
Ah wait I get it. I suppose you could have a down event and a release event in the same update...